dask/array/wrap.py

```diff
@@ -188,7 +188,10 @@ def full(shape, fill_value, *args, **kwargs):
             f"fill_value must be scalar. Received {type(fill_value).__name__} instead."
         )
     if "dtype" not in kwargs:
-        kwargs["dtype"] = type(fill_value)
+        if hasattr(fill_value, "dtype"):
+            kwargs["dtype"] = np.dtype(fill_value.dtype)
```

Comment:
You don't need to wrap the dtype with `np.dtype`

```suggestion
            kwargs["dtype"] = fill_value.dtype
```